src/mpid/ch4/netmod/ofi/ofi_pre.h

```diff
@@ -171,7 +171,9 @@ typedef struct {
 
 typedef struct {
     fi_addr_t dest;
-#if defined MPIDI_OFI_ENABLE_SCALABLE_ENDPOINTS || defined MPIDI_OFI_ENABLE_RUNTIME_CHECKS
+#if MPIDI_OFI_ENABLE_RUNTIME_CHECKS
+    unsigned ep_idx:MPIDI_OFI_MAX_ENDPOINTS_BITS_SCALABLE;
+#elif MPIDI_OFI_ENABLE_SCALABLE_ENDPOINTS
```

Comment:
Yes, this was by design. We wanted to be able to configure things either way because some use cases require this design. Being able to configure at runtime is very helpful for debugging as well as the simplicity of delivering a single binary that can do anything. Being able to make the decisions at compile time is important for performance because all of the `if ()` branches can compile out.

Are you still seeing the problem with the latest patch?